api/storage_service.cc

```diff
@@ -459,12 +459,15 @@ static future<json::json_return_type> describe_ring_as_json(sharded<service::sto
     co_return json::json_return_type(stream_range_as_array(co_await ss.local().describe_ring(keyspace), token_range_endpoints_to_json));
 }
 
-static std::vector<table_id> get_table_ids(const std::vector<table_info>& table_infos) {
-    std::vector<table_id> table_ids{table_infos.size()};
-    boost::transform(table_infos, table_ids.begin(), [] (const auto& ti) {
-        return ti.id;
+static std::vector<::table_info> get_table_infos(const std::vector<table_info>& infos) {
+    std::vector<::table_info> table_infos{infos.size()};
+    boost::transform(infos, table_infos.begin(), [] (const auto& ti) {
+        return ::table_info{
+            .id = ti.id,
+            .name = ti.name
+        };
     });
-    return table_ids;
+    return table_infos;
 }
```

Comment:
Why not use the new `table_info` from `schema_fwd.hh` in `api/` (instead of the `api::` variant) and then this is not necessary.